ql/src/java/org/apache/hadoop/hive/ql/udf/generic/AbstractGenericUDFArrayBase.java

```diff
@@ -49,6 +52,12 @@ protected AbstractGenericUDFArrayBase(String functionName, int minArgCount, int
         this.minArgCount = minArgCount;
         this.maxArgCount = maxArgCount;
         this.outputCategory = outputCategory;
+        this.outputPrimitiveCategory = null;
```

Comment:
This line seems unnecessary. Isnt this `null` by default since it is not initialized?

Reply:
Removed that variable